src/core/src/package_managers/YumPackageManager.py

```diff
@@ -345,28 +370,434 @@ def get_package_size(self, output):
+    def __init_auto_update_for_yum_cron(self):
+        """ Initializes all generic auto OS update variables with the config values for yum cron service """
+        self.os_patch_configuration_settings_file_path = self.yum_cron_configuration_settings_file_path
+        self.download_updates_identifier_text = self.yum_cron_download_updates_identifier_text
+        self.apply_updates_identifier_text = self.yum_cron_apply_updates_identifier_text
+        self.enable_on_reboot_identifier_text = self.yum_cron_enable_on_reboot_identifier_text
+        self.installation_state_identifier_text = self.yum_cron_installation_state_identifier_text
+        self.auto_update_config_pattern_match_text = self.yum_cron_config_pattern_match_text
+        self.enable_on_reboot_check_cmd = self.yum_cron_enable_on_reboot_check_cmd
+        self.install_check_cmd = self.yum_cron_install_check_cmd
+        self.current_auto_os_update_service = Constants.YumAutoOSUpdateServices.YUM_CRON
+
+    def __init_auto_update_for_dnf_automatic(self):
+        """ Initializes all generic auto OS update variables with the config values for dnf automatic service """
+        self.os_patch_configuration_settings_file_path = self.dnf_automatic_configuration_file_path
+        self.download_updates_identifier_text = self.dnf_automatic_download_updates_identifier_text
+        self.apply_updates_identifier_text = self.dnf_automatic_apply_updates_identifier_text
+        self.enable_on_reboot_identifier_text = self.dnf_automatic_enable_on_reboot_identifier_text
+        self.installation_state_identifier_text = self.dnf_automatic_installation_state_identifier_text
+        self.auto_update_config_pattern_match_text = self.dnf_automatic_config_pattern_match_text
+        self.enable_on_reboot_check_cmd = self.dnf_automatic_enable_on_reboot_check_cmd
+        self.install_check_cmd = self.dnf_automatic_install_check_cmd
+        self.current_auto_os_update_service = Constants.YumAutoOSUpdateServices.DNF_AUTOMATIC
+
+    def __init_auto_update_for_packagekit(self):
+        """ Initializes all generic auto OS update variables with the config values for packagekit service """
+        self.os_patch_configuration_settings_file_path = self.packagekit_configuration_file_path
+        self.download_updates_identifier_text = self.packagekit_download_updates_identifier_text
+        self.apply_updates_identifier_text = self.packagekit_apply_updates_identifier_text
+        self.enable_on_reboot_identifier_text = self.packagekit_enable_on_reboot_identifier_text
+        self.installation_state_identifier_text = self.packagekit_installation_state_identifier_text
+        self.auto_update_config_pattern_match_text = self.packagekit_config_pattern_match_text
+        self.enable_on_reboot_check_cmd = self.packagekit_enable_on_reboot_check_cmd
+        self.install_check_cmd = self.packagekit_install_check_cmd
+        self.current_auto_os_update_service = Constants.YumAutoOSUpdateServices.PACKAGEKIT
```

Comment:
These 3 init methods could be merged into one that takes the `current_auto_os_update_service` constant as an argument. Instead of having each command/text as a separate variable, they could all be part of a dict using the `current_auto_os_update_service` constant as the first key, and then the name of the text to get as the second key.

eg. `self.auto_update_config[self.current_auto_os_update_service]["configuration_file_path"] ...`

I am not seeing any differences between the different auto update disabling methods to make use of separation like this.

Reply:
Will address in next PR